src/Service/SoapClient.php

```diff
@@ -0,0 +1,117 @@
+<?php
+
+/**
+ * This file is part of itk-dev/serviceplatformen.
+ *
+ * (c) 2020 ITK Development
+ *
+ * This source file is subject to the MIT license.
+ */
+
+namespace ItkDev\Serviceplatformen\Service;
+
+use Symfony\Component\Cache\Adapter\FilesystemAdapter;
+use Symfony\Component\OptionsResolver\Options;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+use Symfony\Contracts\Cache\CacheInterface;
+use Symfony\Contracts\Cache\ItemInterface;
+
+class SoapClient
+{
+    private array $options;
+
+    public function __construct(array $options)
+    {
+        $resolver = new OptionsResolver();
+        $this->configureOptions($resolver);
+
+        $this->options = $resolver->resolve($options);
+    }
+
+    /**
+     * Executes Soap request.
+     */
+    public function doSoap($url, $request, $action = null, $noCache = false, $cacheKeyOptions = [])
+    {
+        if ($noCache) {
+            return $this->call($url, $request, $action);
+        }
+
+        $cache = $this->getCache();
+
+        $cacheKey = $this->getCacheKey(__METHOD__, [
+            $cacheKeyOptions
+        ]);
+
+        $expirationTime = new \DateTime($this->options['cache_expiration_time']);
```

Comment:
This will not work as expected with `$this->options['cache_expiration_time'] === 'tomorrow 07:00'` and making the call between 00:00 and 07:00.

Reply:
Fixed with inspiration from https://github.com/rimi-itk/timestuff.